src/core/io/src/4C_io_input_file.cpp

```diff
@@ -433,13 +478,36 @@ namespace Core::IO
     }
     else
     {
-      FOUR_C_THROW("Fragment is not a string.");
+      const auto& node = std::get<ryml::ConstNodeRef>(pimpl_->fragment);
+      if (node.is_val())
+      {
+        return std::string_view(node.val().data(), node.val().size());
+      }
+      else
+      {
+        // flatten the yaml node into a string
+        std::ostringstream ss;
+        ss << node;
+        // replace all the yaml markup characters with spaces to receive a dat-style string
+        auto& str = pimpl_->section->as_yaml().dat_style_string;
+        str = ss.str();
+        std::replace(str.begin(), str.end(), '\n', ' ');
+        std::replace(str.begin(), str.end(), ':', ' ');
+        std::replace(str.begin(), str.end(), ',', ' ');
+        std::replace(str.begin(), str.end(), '[', ' ');
+        std::replace(str.begin(), str.end(), ']', ' ');
```

Comment:
```suggestion
        std::replace(str.begin(), str.end(), '[', ' ');
        std::replace(str.begin(), str.end(), ']', ' ');
        std::replace(str.begin(), str.end(), '{', ' ');
        std::replace(str.begin(), str.end(), '}', ' ');
```
Don't we also need these?

Reply:
Once we directly read yaml without converting it to dat internally, testing json explicitly should not be necessary. So not so important for this intermediate step.